paymentsheet/src/main/java/com/stripe/android/paymentsheet/elements/TextFieldController.kt

```diff
@@ -48,11 +49,13 @@ internal class TextFieldController @VisibleForTesting constructor(
     val visibleError: Flow<Boolean> = combine(_fieldState, _hasFocus) { fieldState, hasFocus ->
         fieldState.shouldShowError(hasFocus)
     }
-    override val error: Flow<FieldError?> = visibleError.map { visibleError ->
-        _fieldState.value.getErrorMessageResId()?.let {
-            FieldError(label, it)
-        }?.takeIf { visibleError }
-    }
+    override val error: Flow<FieldError?> = visibleError
+        .filter { it }
+        .map {
+            _fieldState.value.getErrorMessageResId()?.let {
+                FieldError(label, it)
+            }
+        }
```

Comment:
This potentially changes the logic a bit - previously it will emit null when `visibleError` is false, now it just won't emit.

Reply:
think it is ok